studio/components/interfaces/BillingV2/Invoices.tsx

```diff
@@ -1,92 +1,57 @@
 import InvoiceStatusBadge from './InvoiceStatusBadge'
-import { Invoice, InvoiceStatus } from './Invoices.types'
+import { InvoiceStatus } from './Invoices.types'
 
 const PAGE_LIMIT = 10
 
-interface InvoicesProps {
-  projectRef: string
-}
-
-const Invoices = ({ projectRef }: InvoicesProps) => {
+const Invoices = () => {
   const { ui } = useStore()
-  const [loading, setLoading] = useState<any>(false)
-
+  const { ref: projectRef } = useParams()
   const [page, setPage] = useState(1)
-  const [count, setCount] = useState(0)
-  const [invoices, setInvoices] = useState<Invoice[]>([])
 
   const offset = (page - 1) * PAGE_LIMIT
+  const { data: count, isError: isErrorCount } = useProjectInvoicesCountQuery({ projectRef })
+  const { data, error, isLoading, isError, isSuccess } = useProjectInvoicesQuery({
+    projectRef,
+    limit: PAGE_LIMIT,
+    offset,
+  })
+  const invoices = data || []
 
   useEffect(() => {
-    if (!projectRef) return
-
-    let cancel = false
-    const page = 1
-
-    const fetchInvoiceCount = async () => {
-      const res = await head(`${API_URL}/projects/${projectRef}/invoices`, ['X-Total-Count'])
-      if (!cancel) {
-        if (res.error) {
-          ui.setNotification({ category: 'error', message: res.error.message })
-        } else {
-          setCount(res['X-Total-Count'])
-        }
-      }
-    }
-
-    setPage(page)
-    fetchInvoices(page)
-    fetchInvoiceCount()
-
-    return () => {
-      cancel = true
-    }
+    setPage(1)
   }, [projectRef])
 
-  const fetchInvoices = async (page: number) => {
-    setLoading(true)
-    setPage(page)
-
-    const offset = (page - 1) * PAGE_LIMIT
-    const invoices = await get(
-      `${API_URL}/projects/${projectRef}/invoices?offset=${offset}&limit=${PAGE_LIMIT}`
-    )
-
-    if (invoices.error) {
-      ui.setNotification({
-        category: 'error',
-        message: `Failed to fetch invoices: ${invoices.error.message}`,
-      })
-    } else {
-      setInvoices(invoices)
-    }
-
-    setLoading(false)
-  }
-
-  const fetchInvoice = async (invoiceId: string) => {
-    const invoice = await get(`${API_URL}/stripe/invoices/${invoiceId}`)
-    if (invoice?.invoice_pdf) {
-      window.open(invoice.invoice_pdf, '_blank')
-    } else {
+  const fetchInvoice = async (id: string) => {
+    try {
+      const invoice = await getInvoice({ id })
+      if (invoice?.invoice_pdf) window.open(invoice.invoice_pdf, '_blank')
+    } catch (error: any) {
       ui.setNotification({
         category: 'info',
-        message: `Failed to fetch the selected invoice: ${invoice.error.message}`,
+        message: `Failed to fetch the selected invoice: ${error.message}`,
       })
     }
   }
 
   return (
     <div className="container my-4 max-w-4xl space-y-1">
-      <Loading active={loading}>
+      {isLoading && <GenericSkeletonLoader />}
+
+      {isError && <AlertError error={error} subject="Failed to retrieve project invoices" />}
```

Comment:
We should use the new alerts here, right?

Reply:
Ahh yup! AlertError is already using the new Alert component - its just a standardized component on how we want to show UI Alerts across the dashboard - an error message and a quick button to direct users to the support form the title prefilled and error message added to the message body